### PR TITLE
refactoring with constants (oleg.matviiv) #15

### PR DESCRIPTION
**Pull request by Sayok**

### CODE REVIEW

File: homework.rb [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
+VARIANT     = 15
+NUM_RANGE   = 1..999_999
+CHAR_RANGE  = 'AA'..'ZZ'
 
+# It should fit this pattern "123456-ZXCE-ABCD"
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  numbers = -> { rand(NUM_RANGE).to_s.rjust(6, '0') }
+  letters = -> { CHAR_RANGE.to_a.sample(2).join }
```
Comment (sveredyuk) [inline homework.rb]: 2 magic number

File: homework.rb [hunk 2]
```diff
@@ -1,6 +1,11 @@
-VARIANT = 0
+VARIANT     = 15
+NUM_RANGE   = 1..999_999
+CHAR_RANGE  = 'AA'..'ZZ'
 
+# It should fit this pattern "123456-ZXCE-ABCD"
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  numbers = -> { rand(NUM_RANGE).to_s.rjust(6, '0') }
```
Comment (sveredyuk) [inline homework.rb]: 6 ? 0 ? Magic data

File: homework.rb [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
+  letters = -> { CHAR_RANGE.to_a.sample(2).join }
+
+  [numbers.call, letters.call, letters.call].join('-')
```
Comment (sveredyuk) [inline homework.rb]: cool guy style:
```ruby
[numbers, letters, letters].map(&:call).join('-')
```

**Comment by Sayok** [comment]
please check @sveredyuk

**Review by sveredyuk — APPROVED** [review]

**Pull request closed without merging**